packages/core/src/account/simple.ts

```diff
@@ -89,6 +95,48 @@ export class SimpleSmartContractAccount<
     return this.owner.signMessage(msg);
   }
 
+  signTypedData(
+    params: Omit<SignTypedDataParameters, "privateKey">
+  ): Promise<Hash> {
+    return this.owner.signTypedData(params);
+  }
+
+  async signMessageWith6492(msg: string | Uint8Array): Promise<`0x${string}`> {
+    const [isDeployed, signature] = await Promise.all([
+      this.isAccountDeployed(),
+      this.signMessage(msg),
+    ]);
+
+    if (isDeployed) {
+      return signature;
+    }
+
+    return wrapWith6492({
+      factoryAddress: this.factoryAddress,
+      initCode: await this.getAccountInitCode(),
+      signature,
+    });
+  }
+
+  async signTypedDataWith6492(
+    params: Omit<SignTypedDataParameters, "privateKey">
+  ): Promise<Hash> {
+    const [isDeployed, signature] = await Promise.all([
+      this.isAccountDeployed(),
+      this.signTypedData(params),
+    ]);
+
+    if (isDeployed) {
+      return signature;
+    }
+
+    return wrapWith6492({
+      factoryAddress: this.factoryAddress,
+      initCode: await this.getAccountInitCode(),
+      signature,
+    });
+  }
```

Comment:
ugh oh wait... simple account doesn't support 1271 signature validation :/ so this won't work